server/internal/server/vector_stores.go

```diff
@@ -0,0 +1,321 @@
+			}
+		}
+	}
+	for _, cm := range cms {
+		if _, ok := req.Metadata[cm.Key]; !ok {
+			if err := s.store.DeleteCollectionMetadata(cm.CollectionID, cm.Key); err != nil {
+				return nil, status.Errorf(codes.Internal, "delete collection metadata: %s", err)
+			}
+		}
+	}
+	cms, err = s.store.ListCollectionMetadataByCollectionID(c.CollectionID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "list collection metadata: %s", err)
+	}
+
+	// update collection
+	if req.Name != "" && req.Name != c.Name {
+		if err := s.vstoreClient.UpdateVectorStoreName(ctx, c.Name, req.Name); err != nil {
+			return nil, status.Errorf(codes.Internal, "update vector store: %s", err)
+		}
+		c.Name = req.Name
+	}
+	if req.ExpiresAfter != nil {
+		c.Anchor = store.ExpiresAfterAnchor(req.ExpiresAfter.Anchor)
+		c.ExpiresAfterDays = req.ExpiresAfter.Days
+	}
+	if err := s.store.UpdateCollection(c); err != nil {
+		return nil, status.Errorf(codes.Internal, "update collection: %s", err)
+	}
+
+	return toVectorStoreProto(c, cms), nil
+}
+
+// DeleteVectorStore deletes a vector store.
+func (s *S) DeleteVectorStore(
+	ctx context.Context,
+	req *v1.DeleteVectorStoreRequest,
+) (*v1.DeleteVectorStoreResponse, error) {
+	userInfo, err := s.extractUserInfoFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := getCollectionID(req.VectorStoreId, true)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.vstoreClient.DeleteVectorStore(ctx, req.VectorStoreId); err != nil {
+		return nil, status.Errorf(codes.Internal, "delete vector store: %s", err)
+	}
+	if err := s.store.DeleteCollection(userInfo.ProjectID, id); err != nil {
+		return nil, status.Errorf(codes.Internal, "delete collection: %s", err)
+	}
+	if err := s.store.DeleteCollectionMetadataByCollectionID(id); err != nil {
+		return nil, status.Errorf(codes.Internal, "delete collection metadata: %s", err)
+	}
+	return &v1.DeleteVectorStoreResponse{
+		Id:      req.VectorStoreId,
+		Object:  vectorStoreObject,
+		Deleted: true,
+	}, nil
+}
+
+func toVectorStoreProto(c *store.Collection, cms []*store.CollectionMetadata) *v1.VectorStore {
+	m := map[string]string{}
+	for _, cm := range cms {
+		m[cm.Key] = cm.Value
+	}
+
+	return &v1.VectorStore{
+		Id:         fmt.Sprintf("%d", c.CollectionID),
+		Object:     vectorStoreObject,
+		CreatedAt:  c.CreatedAt.Unix(),
+		Name:       c.Name,
+		UsageBytes: c.UsageBytes,
+		FileCounts: &v1.VectorStore_FileCounts{
+			InProgress: c.FileCountsInProgress,
+			Completed:  c.FileCountsCompleted,
+			Failed:     c.FileCountsFailed,
+			Cancelled:  c.FileCountsCancelled,
+			Total:      c.FileCountsTotal,
+		},
+		Status: string(c.Status),
+		ExpiresAfter: &v1.ExpiresAfter{
+			Anchor: string(c.Anchor),
+			Days:   c.ExpiresAfterDays,
+		},
+		ExpiresAt:    c.ExpiresAt,
+		LastActiveAt: c.LastActiveAt,
+		Metadata:     m,
+	}
+}
+
+func getCollectionID(vid string, required bool) (int64, error) {
```

Comment:
How about removing the `required` parameter? I think that simplifies the function interface a lot. I believe we only need a special logic in `ListVectorStores`

```go
var afterID string
if req.After != "" {
        var err error
        afterID, err = getCollectionID(req.After)
	if err != nil {
		return nil, err
	}
}
```